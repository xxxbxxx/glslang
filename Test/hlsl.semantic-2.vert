struct S2 {
    float2 UV;
};

struct S {
    S2    t	: TEXCOORD0;
};

S main(S ins)
{
    S s;
    return s;
}
