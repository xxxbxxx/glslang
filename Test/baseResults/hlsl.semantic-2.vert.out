hlsl.semantic-2.vert
Shader version: 500
0:? Sequence
0:10  Function Definition: @main(struct-S-struct-S2-vf211; ( temp structure{ temp structure{ temp 2-component vector of float UV} : TEXCOORD0 t})
0:10    Function Parameters: 
0:10      'ins' ( in structure{ temp structure{ temp 2-component vector of float UV} : TEXCOORD0 t})
0:?     Sequence
0:12      Branch: Return with expression
0:12        's' ( temp structure{ temp structure{ temp 2-component vector of float UV} : TEXCOORD0 t})
0:10  Function Definition: main( ( temp void)
0:10    Function Parameters: 
0:?     Sequence
0:10      Sequence
0:10        move second child to first child ( temp 2-component vector of float)
0:10          UV: direct index for structure ( temp 2-component vector of float)
0:10            t: direct index for structure ( temp structure{ temp 2-component vector of float UV} : TEXCOORD0)
0:?               'ins' ( temp structure{ temp structure{ temp 2-component vector of float UV} : TEXCOORD0 t})
0:10              Constant:
0:10                0 (const int)
0:10            Constant:
0:10              0 (const int)
0:?           'ins.t.UV' (layout( location=0) in 2-component vector of float : TEXCOORD0)
0:10      Sequence
0:10        move second child to first child ( temp 2-component vector of float)
0:?           '@entryPointOutput.t.UV' (layout( location=0) out 2-component vector of float : TEXCOORD0)
0:10          UV: direct index for structure ( temp 2-component vector of float)
0:10            t: direct index for structure ( temp structure{ temp 2-component vector of float UV} : TEXCOORD0)
0:10              Function Call: @main(struct-S-struct-S2-vf211; ( temp structure{ temp structure{ temp 2-component vector of float UV} : TEXCOORD0 t})
0:?                 'ins' ( temp structure{ temp structure{ temp 2-component vector of float UV} : TEXCOORD0 t})
0:10              Constant:
0:10                0 (const int)
0:10            Constant:
0:10              0 (const int)
0:?   Linker Objects
0:?     '@entryPointOutput.t.UV' (layout( location=0) out 2-component vector of float : TEXCOORD0)
0:?     'ins.t.UV' (layout( location=0) in 2-component vector of float : TEXCOORD0)


Linked vertex stage:


Shader version: 500
0:? Sequence
0:10  Function Definition: @main(struct-S-struct-S2-vf211; ( temp structure{ temp structure{ temp 2-component vector of float UV} : TEXCOORD0 t})
0:10    Function Parameters: 
0:10      'ins' ( in structure{ temp structure{ temp 2-component vector of float UV} : TEXCOORD0 t})
0:?     Sequence
0:12      Branch: Return with expression
0:12        's' ( temp structure{ temp structure{ temp 2-component vector of float UV} : TEXCOORD0 t})
0:10  Function Definition: main( ( temp void)
0:10    Function Parameters: 
0:?     Sequence
0:10      Sequence
0:10        move second child to first child ( temp 2-component vector of float)
0:10          UV: direct index for structure ( temp 2-component vector of float)
0:10            t: direct index for structure ( temp structure{ temp 2-component vector of float UV} : TEXCOORD0)
0:?               'ins' ( temp structure{ temp structure{ temp 2-component vector of float UV} : TEXCOORD0 t})
0:10              Constant:
0:10                0 (const int)
0:10            Constant:
0:10              0 (const int)
0:?           'ins.t.UV' (layout( location=0) in 2-component vector of float : TEXCOORD0)
0:10      Sequence
0:10        move second child to first child ( temp 2-component vector of float)
0:?           '@entryPointOutput.t.UV' (layout( location=0) out 2-component vector of float : TEXCOORD0)
0:10          UV: direct index for structure ( temp 2-component vector of float)
0:10            t: direct index for structure ( temp structure{ temp 2-component vector of float UV} : TEXCOORD0)
0:10              Function Call: @main(struct-S-struct-S2-vf211; ( temp structure{ temp structure{ temp 2-component vector of float UV} : TEXCOORD0 t})
0:?                 'ins' ( temp structure{ temp structure{ temp 2-component vector of float UV} : TEXCOORD0 t})
0:10              Constant:
0:10                0 (const int)
0:10            Constant:
0:10              0 (const int)
0:?   Linker Objects
0:?     '@entryPointOutput.t.UV' (layout( location=0) out 2-component vector of float : TEXCOORD0)
0:?     'ins.t.UV' (layout( location=0) in 2-component vector of float : TEXCOORD0)

// Module Version 10000
// Generated by (magic number): 80001
// Id's are bound by 33

                              Capability Shader
               1:             ExtInstImport  "GLSL.std.450"
                              MemoryModel Logical GLSL450
                              EntryPoint Vertex 4  "main" 23 28
                              Source HLSL 500
                              Name 4  "main"
                              Name 8  "S2"
                              MemberName 8(S2) 0  "UV"
                              Name 9  "S"
                              MemberName 9(S) 0  "t"
                              Name 13  "@main(struct-S-struct-S2-vf211;"
                              Name 12  "ins"
                              Name 15  "s"
                              Name 19  "ins"
                              Name 23  "ins.t.UV"
                              Name 28  "@entryPointOutput.t.UV"
                              Name 29  "param"
                              Decorate 23(ins.t.UV) Location 0
                              Decorate 28(@entryPointOutput.t.UV) Location 0
               2:             TypeVoid
               3:             TypeFunction 2
               6:             TypeFloat 32
               7:             TypeVector 6(float) 2
           8(S2):             TypeStruct 7(fvec2)
            9(S):             TypeStruct 8(S2)
              10:             TypePointer Function 9(S)
              11:             TypeFunction 9(S) 10(ptr)
              20:             TypeInt 32 1
              21:     20(int) Constant 0
              22:             TypePointer Input 7(fvec2)
    23(ins.t.UV):     22(ptr) Variable Input
              25:             TypePointer Function 7(fvec2)
              27:             TypePointer Output 7(fvec2)
28(@entryPointOutput.t.UV):     27(ptr) Variable Output
         4(main):           2 Function None 3
               5:             Label
         19(ins):     10(ptr) Variable Function
       29(param):     10(ptr) Variable Function
              24:    7(fvec2) Load 23(ins.t.UV)
              26:     25(ptr) AccessChain 19(ins) 21 21
                              Store 26 24
              30:        9(S) Load 19(ins)
                              Store 29(param) 30
              31:        9(S) FunctionCall 13(@main(struct-S-struct-S2-vf211;) 29(param)
              32:    7(fvec2) CompositeExtract 31 0 0
                              Store 28(@entryPointOutput.t.UV) 32
                              Return
                              FunctionEnd
13(@main(struct-S-struct-S2-vf211;):        9(S) Function None 11
         12(ins):     10(ptr) FunctionParameter
              14:             Label
           15(s):     10(ptr) Variable Function
              16:        9(S) Load 15(s)
                              ReturnValue 16
                              FunctionEnd
